app/src/renderer/connectors/lcdClientMock.js

```diff
@@ -891,6 +916,142 @@ module.exports = {
+  }) {
+    let results = []
+    let fromKey = state.keys.find(a => a.name === name)
+    let fromAccount = state.accounts[fromKey.address]
+    if (fromAccount == null) {
+      results.push(txResult(1, `Nonexistent account`))
+      return results
+    }
+    // check nonce
+    if (parseInt(fromAccount.sequence) !== parseInt(sequence)) {
+      results.push(
+        txResult(
+          2,
+          `Expected sequence "${fromAccount.sequence}", got "${sequence}"`
+        )
+      )
+      return results
+    }
+
+    let proposal = state.proposals.find(
+      proposal => proposal.proposal_id === proposal_id
+    )
+    if (!proposal) {
+      results.push(txResult(3, `Nonexistent proposal`))
+      return results
+    } else if (
+      proposal.proposal_status != `Pending` &&
+      proposal.proposal_status != `Active`
+    ) {
+      results.push(txResult(3, `Proposal #${proposal_id} already finished`))
+      return results
+    }
+
+    // update depositer's balance
+    let coin
+    let submittedDeposit = {
+      proposal_id,
+      depositer,
+      amount
+    }
+
+    // javascript's forEach doesn't support break, so using classic for loop...
+    for (let i = 0; i < amount.length; i++) {
+      coin = amount[i]
+      let depositCoinAmt = parseInt(coin.amount)
+      let coinBalance = fromAccount.coins.find(c => c.denom === coin.denom)
+
+      if (depositCoinAmt < 0) {
+        results.push(txResult(1, `Amount of ${coin.denom}s cannot be negative`))
+        return results
+      } else if (!coinBalance || coinBalance.amount < depositCoinAmt) {
+        results.push(txResult(1, `Not enough ${coin.denom}s in your account`))
+        return results
+      }
+
+      coinBalance.amount -= depositCoinAmt
+
+      // ============= TOTAL PROPOSAL's DEPOSIT =============
+      // Increment total deposit of the proposal
+      let depositIndex = proposal.total_deposit.findIndex(
+        deposit => deposit.denom === coin.denom
+      )
+      if (depositIndex === -1) {
+        // if there's no previous deposit in that denom we just append it to the total deposit
+        proposal.total_deposit.push(coin)
+      } else {
+        // if there's an existing deposit with that denom we add the submited deposit amount to it
+        let newAmt = String(
+          parseInt(proposal.total_deposit[depositIndex].amount) +
+            parseInt(coin.amount)
+        )
+        proposal.total_deposit[depositIndex].amount = newAmt
+      }
+
+      // ============= USER'S DEPOSITS =============
+      // check if there's an existing deposit by the depositer
+      let prevDepositIndex = state.deposits[proposal_id].findIndex(
+        deposit => deposit.depositer === depositer
+      )
+
+      if (prevDepositIndex === -1) {
+        // if no previous deposit by the depositer, we add it to the existing deposits
+        state.deposits[proposal_id].push(submittedDeposit)
+        break // break since no need to iterate over other coins
```

Comment:
As this is the only break, you could also loop once through all deposits checking for an existing deposit. This would reduce the if then elses in the second "update deposits" loop.

Reply:
yeah I started using `.find()` but then I changed Idk, why though... updated